src/middlewared/middlewared/plugins/system.py

```diff
@@ -64,6 +69,22 @@ def ready(self):
             stdout=subprocess.PIPE,
         )).communicate())[0].decode().strip() or None
 
+        license = get_license()[0]
+        if license:
+            if license.contract_end > date.today():
+                days = (license.contract_end - date.today()).days
+                license = _('%1s contract, expires at %2s, %3d %4s left' % (
+                    _(ContractType(license.contract_type).name.title()),
+                    license.contract_end.strftime("%x"),
+                    days,
+                    ungettext('day', 'days', days),
+                ))
+            else:
+                license = _('%1s contract, expired at %2s' % (
+                    _(ContractType(license.contract_type).name.title()),
+                    license.contract_end.strftime("%x"),
+                ))
```

Comment:
I am not a fan of providing a human readable output for `license`. Any way we can provide a dict with the desired data and format that on the UI side?